python/cudf/cudf/core/multiindex.py

```diff
@@ -885,6 +885,27 @@ def difference(self, other, sort=None):
             temp_other = self.to_pandas()
         return temp_self.difference(temp_other, sort)
 
+    def append(self, other):
+        """
+        Append a collection of Index options together
+
+        Parameters
+        ----------
+        other : Index or list/tuple of indices
+
+        Returns
+        -------
+        appended : Index
+        """
+        if isinstance(other, (list, tuple)):
+            other = [
+                o.to_pandas() if hasattr(o, "to_pandas") else o for o in other
+            ]
+        elif hasattr(other, "to_pandas"):
+            other = other.to_pandas()
+
+        return cudf.from_pandas(self.to_pandas().append(other))
```

Comment:
Can you explain these two branches?

If `other` is a list or a tuple what's the expected types in the list / tuple?

if `other` is a MultiIndex object, it looks like we copy it back to host.

Regardless, we copy `self` to host, append the objects on the host, and then copy back to GPU.

Why can't we just do the append on the GPU via `concat`?

Reply:
> If `other` is a list or a tuple what's the expected types in the list / tuple?
Always has to a list/tuple of `MultiIndex` objects. This is because we cannot currently support appending any other type to `MultiIndex` object for which pandas results in nested Index objects(and we only support 1-d columns).

> if `other` is a MultiIndex object, it looks like we copy it back to host.
Yes

> Regardless, we copy `self` to host, append the objects on the host, and then copy back to GPU.
Correct, Previous code was relying on pandas API for broader support of appending objects of other types. But after some investigation, I realized that we will anyways fail while importing a multi-dimensional column(as explained above) in `as_column`. So hence mandating `MultiIndex.append` to take in `MultiIndex` objects.

> Why can't we just do the append on the GPU via `concat`?
Changed the code to perform append on GPU itself 👍 , didn't know we had `MultiIndex._concat` already. However, it needed some changes to match panda's behavior.